dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java

```diff
@@ -359,6 +361,68 @@ public void testGitHubReleaseNoWorkflowOnDockstoreNoUser() throws Exception {
         }
     }
 
+    /**
+     * Tests:
+     * An unpublished workflow with invalid versions can have its descriptor type changed
+     * The workflow can then have new valid versions registered
+     * The valid workflow cannot have its descriptor type changed anymore (because it's valid)
+     * The published workflow cannot have its descriptor type changed anymore (because it's published)
+     * @throws Exception    DB problem
+     */
+    @Test
+    public void testDescriptorChange() throws Exception {
+        CommonTestUtilities.cleanStatePrivate2(SUPPORT, false);
+        final ApiClient webClient = getWebClient(BasicIT.USER_2_USERNAME, testingPostgres);
+        io.dockstore.openapi.client.ApiClient openAPIWebClient = getOpenAPIWebClient(BasicIT.USER_2_USERNAME, testingPostgres);
+        io.dockstore.openapi.client.api.WorkflowsApi workflowsApi = new io.dockstore.openapi.client.api.WorkflowsApi(openAPIWebClient);
+        WorkflowsApi client = new WorkflowsApi(webClient);
+        List<Workflow> workflows = client.handleGitHubRelease(workflowRepo, BasicIT.USER_2_USERNAME, "refs/heads/missingPrimaryDescriptor", installationId);
+        assertEquals("Should only have one service", 1, workflows.size());
+        // Ensure that new workflow is created and is what is expected
+        Workflow workflow = client.getWorkflowByPath("github.com/" + workflowRepo + "/foobar", "", false);
+        assertEquals("Should be a WDL workflow", Workflow.DescriptorTypeEnum.WDL, workflow.getDescriptorType());
+        assertEquals("Should be type DOCKSTORE_YML", Workflow.ModeEnum.DOCKSTORE_YML, workflow.getMode());
+        assertEquals("Should have one version", 1, workflow.getWorkflowVersions().size());
+        assertFalse("Should be invalid (wrong language, bad version)", workflow.getWorkflowVersions().get(0).isValid());
+
+        workflowsApi.updateDescriptorType(workflow.getId(), DescriptorLanguage.CWL.toString());
+        io.dockstore.openapi.client.model.Workflow updatedWorkflowAfterModifyingDescriptorType = workflowsApi.getWorkflow(workflow.getId(), "");
+        Assert.assertEquals("The descriptor language should have been changed", io.dockstore.openapi.client.model.Workflow.DescriptorTypeEnum.CWL, updatedWorkflowAfterModifyingDescriptorType.getDescriptorType());
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=6225903)